### PR TITLE
Feature password accordion

### PR DESCRIPTION
**Pull request by iam-robin**
## Proposed Changes

  - add password input logic into form accordion
  - add password criteria validation
  - add new styles for data protection checkbox
  - clean up registration component

### CODE REVIEW

File: src/components/registration/Registration.tsx [hunk 1]
```diff
@@ -592,20 +474,6 @@ const Registration = () => {
 										'registration.required.infoText'
```
Comment (koepferd) [inline src/components/registration/Registration.tsx:463]: Is there a reason why you only moved a part of the submit section up to the accordion?
Please move comment and infoText up to the submit button.

File: src/components/registration/RegistrationPassword.tsx [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+import './registrationPassword.styles';
+
+interface RegistrationPasswordProps {
+	onPasswordChange: Function;
+	onValidityChange: Function;
+}
+
+export const RegistrationPassword = (props: RegistrationPasswordProps) => {
+	const [isValid, setIsValid] = useState<AccordionItemValidity>('initial');
+
```
Comment (koepferd) [inline src/components/registration/RegistrationPassword.tsx]: please remove empty line

File: src/components/registration/RegistrationPassword.tsx [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+	};
+
+	return (
+		<div className="registrationPassword">
+			<Text
+				text={translate('registration.password.intro')}
+				type="infoLargeAlternative"
+			/>
+			<ul className="registrationPassword__validation">
+				<li
```
Comment (koepferd) [inline src/components/registration/RegistrationPassword.tsx]: You can use an array here to render these list items with their related texts and validation.

**Review by koepferd — CHANGES_REQUESTED** [review]

File: src/components/registration/Registration.tsx [hunk 4]
```diff
@@ -541,16 +433,6 @@ const Registration = () => {
...
-									item={inputItemPassword}
-									inputHandle={handlepasswordChange}
-								/>
-								<InputField
-									item={inputItemPasswordConfirmation}
-									inputHandle={
-										handlePasswordConfirmationChange
-									}
-								/>
 								{consultingType === 1 && (
```
Comment (koepferd) [inline src/components/registration/Registration.tsx]: This special logic for ct 1 needs to be added to the password component in the accordion

File: src/components/registration/Registration.tsx (hunk 1 above)
Comment (iam-robin) [reply src/components/registration/Registration.tsx:463]: clean up of the mandatory info text and * will be done in the voluntary-info-story of the current sprint

**Review by koepferd — APPROVED** [review]

**Pull request merged**